c7n/resources/fsx.py

```diff
@@ -332,6 +337,94 @@ class KmsFilterFsxBackup(KmsRelatedFilter):
+                resource: fsx
+                filters:
+                  - type: consecutive-backups
+                    days: 5
+                actions:
+                  - notify
+    """
+    schema = type_schema('consecutive-backups',
+                         days={'type': 'number', 'minimum': 1},
+                         required=['days'])
+    permissions = ('fsx:DescribeBackups', 'fsx:DescribeVolumes',)
+    annotation = 'c7n:FSxBackups'
+
+    def describe_backups(self, client, name=None, filters=[]):
+        desc_backups = []
+        try:
+            paginator = client.get_paginator('describe_backups')
+            paginator.PAGE_ITERATOR_CLS = RetryPageIterator
+            desc_backups = paginator.paginate(Filters=[
+                {
+                    'Name': name,
+                    'Values': filters,
+                }]).build_full_result().get('Backups', [])
+        except Exception as err:
+            self.log.warning(
+                'Unable to describe backups for ids: %s - %s' % (filters, err))
+        return desc_backups
+
+    def process_resource_set(self, client, resources):
+        ontap_fids, nonontap_fids = [], []
+        fid_backups = {}
+
+        for r in resources:
+            if r['FileSystemType'] == 'ONTAP':
+                ontap_fids.append(r['FileSystemId'])
+            else:
+                nonontap_fids.append(r['FileSystemId'])
+        if ontap_fids:
+            ontap_volumes = client.describe_volumes(Filters=[
+                {
+                    'Name': 'file-system-id',
+                    'Values': ontap_fids,
+                }])
+            ontap_vids = [v['VolumeId'] for v in ontap_volumes['Volumes']]
+            ontap_backups = self.describe_backups(client, 'volume-id', ontap_vids)
+            if ontap_backups:
+                for ontap in ontap_backups:
+                    fid_backups.setdefault(ontap['Volume']
+                                           ['FileSystemId'], []).append(ontap)
+        if nonontap_fids:
+            nonontap_backups = self.describe_backups(client, 'file-system-id', nonontap_fids)
+            if nonontap_backups:
+                for nonontap in nonontap_backups:
+                    fid_backups.setdefault(nonontap['FileSystem']
+                                           ['FileSystemId'], []).append(nonontap)
+        for r in resources:
+            r[self.annotation] = fid_backups.get(r['FileSystemId'], [])
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('fsx')
+        results = []
+        retention = self.data.get('days')
+        utcnow = datetime.utcnow()
+        expected_dates = set()
+        for days in range(1, retention + 1):
+            expected_dates.add((utcnow - timedelta(days=days)).strftime('%Y-%m-%d'))
+
+        for resource_set in chunks(
+                [r for r in resources if self.annotation not in r], 50):
```

Comment:
Actually I noticed something here that we'll need to tweak. First, it looks like the max number of filter values per API call is 20 ([reference](https://github.com/boto/botocore/blob/9577af2dde9e9efef48c0ac95bd8d87c214b14bf/botocore/data/fsx/2018-03-01/service-2.json#L3004-L3033)). And second, we'll need to handle the chunking at a different level. As-is the chunking includes ontap and non-ontap filesystems - we'd want to chunk those separately because they feed into separate API calls. Also since ontap backups are volume-based and there may be more than one volume per filesystem, we'll need to chunk at the volume-level rather than filesystem-level.

My gut says that it'd be helpful to:
* Split `process_resource_set` into dedicated separate methods for ontap and non-ontap resources
* Handle the chunking in those individual methods, at the volume level for ontap and filesystem level for non-ontap

Reply:
- As suggested I have splitted the method `process_resource_set` into two as for ontap and non-ontap resources
- Changed the chunking logics as well. 
- Now it should handle the describe filter limitation issue.